stylecop.ruleset

```diff
@@ -30,5 +30,9 @@
     <Rule Id="SA1602" Action="None" />
     <Rule Id="SA1649" Action="None" />
     <Rule Id="SA1652" Action="None" />
+    <Rule Id="SA1414" Action="None" />
```

Comment:
why disable SA1414, SA1316, and SA1142.  I think those rules will increase readability.

Reply:
I agree with you, but here is the list of all the changes we would need to make:

```
BackgroundTask.cs(13,31): error SA1316: Tuple element names should use correct casing [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
BackgroundTask.cs(13,67): error SA1316: Tuple element names should use correct casing [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(34,24): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(34,49): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
ShutdownHandler.cs(17,48): error SA1316: Tuple element names should use correct casing [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
ShutdownHandler.cs(17,74): error SA1316: Tuple element names should use correct casing [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
ShutdownHandler.cs(17,101): error SA1316: Tuple element names should use correct casing [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(30,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(30,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(37,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(37,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(37,43): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(45,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(45,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(45,43): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(45,47): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(54,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(54,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(54,43): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(54,47): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(54,51): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,43): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,47): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,51): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,55): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,59): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(87,63): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,43): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,47): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,51): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,55): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,59): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,63): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(100,67): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(75,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(75,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(75,43): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(75,47): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(75,51): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(75,55): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(75,59): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(64,35): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(64,39): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(64,43): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(64,47): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(64,51): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
TaskEx.cs(64,55): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
LinqEx.cs(71,36): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
LinqEx.cs(71,42): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(86,24): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(86,30): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
LinqEx.cs(90,123): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
LinqEx.cs(90,129): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(338,26): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(338,44): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(341,26): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
CertificateHelper.cs(341,44): error SA1414: Tuple types in signatures should have element names [/home/als5ev/Desktop/debug/iotedge/edge-util/src/Microsoft.Azure.Devices.Edge.Util/Microsoft.Azure.Devices.Edge.Util.csproj]
```

I would rather handle this in a separate PR to not add the noise here.

Reply:
@philipktlin If you disagree I can make the change here.

Reply:
Talked with @philipktlin and we decided it is better to address in a separate PR.